src/codec.rs

```diff
@@ -635,8 +635,14 @@ macro_rules! impl_array {
 							$dest.write(&typed)
 						}};
 						( $ty:ty, $self:ident, $dest:ident ) => {{
-							let typed = unsafe { mem::transmute::<&[T], &[$ty]>(&$self[..]) };
-							$dest.write(<[$ty] as AsByteSlice<$ty>>::as_byte_slice(typed))
+							if cfg!(target_endian = "little") {
+								let typed = unsafe { mem::transmute::<&[T], &[$ty]>(&$self[..]) };
+								$dest.write(<[$ty] as AsByteSlice<$ty>>::as_byte_slice(typed))
+							} else {
+								for item in self.iter() {
+									item.encode_to(dest);
+								}
```

Comment:
We can probably do a bit better than this by reading in little endian and then reversing bytes for big endian inplace. But I'm not sure of the speed improvment and I'm not sure of the need. Thus I kept it simple for now.